Lib/pickletools.py

```diff
@@ -565,6 +565,41 @@ def read_bytes8(f):
               the number of bytes, and the second argument is that many bytes.
               """)
 
+
+def read_bytearray8(f):
+    r"""
+    >>> import io, struct, sys
+    >>> read_bytearray8(io.BytesIO(b"\x00\x00\x00\x00\x00\x00\x00\x00abc"))
+    bytearray(b'')
+    >>> read_bytearray8(io.BytesIO(b"\x03\x00\x00\x00\x00\x00\x00\x00abcdef"))
+    bytearray(b'abc')
+    >>> bigsize8 = struct.pack("<Q", sys.maxsize//3)
+    >>> read_bytearray8(io.BytesIO(bigsize8 + b"abcdef"))  #doctest: +ELLIPSIS
+    Traceback (most recent call last):
+    ...
+    ValueError: expected ... bytes in a bytearray8, but only 6 remain
+    """
+
+    n = read_uint8(f)
+    assert n >= 0
+    if n > sys.maxsize:
+        raise ValueError("bytearray8 byte count > sys.maxsize: %d" % n)
```

Comment:
The test is fine, but I hope that it will never fail on any system :-D 255 greater than sys.maxsize, hum? :-)

Reply:
Who knows, with MicroPython ;-)

But seriously, `read_uint8` reads a 64-bit integer.

Reply:
Oh. I read 8 as 8 bits :-) Ok, in that case the if totally makes sense ;-)